controllers/azuremachine_controller.go

```diff
@@ -152,28 +157,71 @@ func (r *AzureMachineReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, ret
 		return ctrl.Result{}, nil
 	}
 
-	azureClusterName := client.ObjectKey{
+	controlPlaneRefKind := cluster.Spec.ControlPlaneRef.Kind
+	controlPlaneRefGV, err := schema.ParseGroupVersion(cluster.Spec.ControlPlaneRef.APIVersion)
+	if err != nil {
+		return ctrl.Result{}, errors.Wrap(err, "failed to parse gv from control plane apiversion")
+	}
+
+	infraRefKind := cluster.Spec.InfrastructureRef.Kind
+	infraRefGV, err := schema.ParseGroupVersion(cluster.Spec.InfrastructureRef.APIVersion)
+	if err != nil {
+		return ctrl.Result{}, errors.Wrap(err, "failed to parse gv from infra ref apiversion")
+	}
+
+	// AzureManagedControlPlane only works with AzureManagedCluster
+	var isManaged bool
+	clusterDescriberRef := cluster.Spec.InfrastructureRef
```

Comment:
feeling the downside of making `AzureManagedCluster` a no-op right about now...(we could just use infra ref otherwise)